### PR TITLE
Mine page: add the option to edit and delete expense

### PR DESCRIPTION
**Pull request by ShellyGolden**
blocked by https://github.com/redhat-beyond/Xpense/pull/83 
relevant issue: https://github.com/redhat-beyond/Xpense/issues/93

image

after delete:
image

edit page:
image

### CODE REVIEW

**Comment by LiorKGOW** [comment]
#111 was merged
Could you update the top comment?

**Comment by ShellyGolden** [comment]
> #111 was merged Could you update the top comment?

fixed
thanks

File: house/views.py [hunk 1]
```diff
@@ -94,4 +94,25 @@ def house_create(request):
     else:
         house_form = HouseCreationForm()
 
-    return render(request, HOUSE_CREATE_ROUTE, {'form': house_form, 'msg': errorMsg})
+    return render(request, HOUSE_CREATE_ROUTE, {'form': house_form, 'msg': error_message})
+
+
+@login_required
+def edit_expense(request, id):
+    expense = get_object_or_404(Expenses, pk=id)
```
Comment (ronyyosef) [inline house/views.py:102]: Please add a condition that make sure that the expenses is really on of the user expenses

File: house/views.py [hunk 2]
```diff
@@ -94,4 +94,25 @@ def house_create(request):
...
+            expense = form.save(commit=False)
+            expense.save()
+            return HttpResponseRedirect('/../house')
+    else:
+        return render(request, MINE_EDIT_EXPENSE_ROUTE, context)
+
+
+@login_required
+def delete_expense(request, id):
+    expense = get_object_or_404(Expenses, pk=id)
```
Comment (ronyyosef) [inline house/views.py:116]: same here

File: tests/my_house_tests.py [hunk 3]
```diff
@@ -92,6 +102,26 @@ def test_not_add_expenses_to_my_house_with_bad_description(self, client, new_use
...
+    def test_edit_expenses_of_my_house(self, client, new_user):
+        client.force_login(new_user)
+        house = HouseFactory(user=new_user)
+        house.save()
+        ExpenseFactory(house=house, month=1).save()
+        response = client.post('/house/edit_expense/1/', OTHER_EXPENSE_FORM_DATA)
+        assert len(Expenses.objects.all()) == 1
+        assert response.status_code == 302
+        assert response.url == '/../house'
+
```
Comment (Yoavsc129) [inline tests/my_house_tests.py:114]: You should also assert that the expense was actually changed

File: tests/my_house_tests.py [hunk 4]
```diff
@@ -103,3 +133,38 @@ def test_get_add_expense_view(self, client, new_user):
...
```
Comment (Yoavsc129) [inline tests/my_house_tests.py:149]: This test is repeated

**Review by Yoavsc129 — CHANGES_REQUESTED** [review]

File: tests/my_house_tests.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+@pytest.fixture
+def generate_expense_form():
+    form = ExpenseForm(EXPENSE_FORM_DATA)
+    return form
+
+
+@pytest.fixture
+def generate_other_expense_form():
+    form = ExpenseForm(OTHER_EXPENSE_FORM_DATA)
+    return form
```
Comment (shlior7) [inline tests/my_house_tests.py:33]: I don't see any place you are using this fixture, is this for future testing?
if so are we sure we need it as a fixture?

File: house/views.py (hunk 1 above)
Comment (ShellyGolden) [reply house/views.py:105]: fixed

File: house/views.py (hunk 2 above)
Comment (ShellyGolden) [reply house/views.py:125]: fixed

File: tests/my_house_tests.py (hunk 4 above)
Comment (ShellyGolden) [reply tests/my_house_tests.py]: fixed

File: tests/house_tests.py [hunk 6]
```diff
@@ -108,6 +108,8 @@ def test_profession_either_parent(self):
         house2.save()
         houses = _filter_houses_by_form(form_data_filter_tests(), House.objects.all())
         assert len(houses) == 2
+        assert House.objects.get(parent_profession_1=Job.TEACHER) in houses
+        assert House.objects.get(parent_profession_2=Job.TEACHER) in houses
```
Comment (adiabramovitch) [inline tests/house_tests.py:112]: Why is this here?

File: tests/my_house_tests.py (hunk 3 above)
Comment (ShellyGolden) [reply tests/my_house_tests.py:160]: fixed

File: tests/house_tests.py (hunk 6 above)
Comment (ShellyGolden) [reply tests/house_tests.py]: I accidentally deleted these lines while performing a rebase, so I returned them

File: tests/my_house_tests.py [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+    def test_bad_form_description(self, generate_bad_description_expense_form):
+        form = generate_bad_description_expense_form
+        assert not form.is_valid()
+
+
+@pytest.mark.django_db
+class TestMyHouseViewsAndTemplates:
+    def test_get_house_view_function(self, client, new_user):
+        client.force_login(new_user)
```
Comment (Yarboa) [inline tests/my_house_tests.py:63]: Genereric err in tests
Look at the UI slides, should be done in @fixture

File: tests/my_house_tests.py [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+        HouseFactory(user=new_user).save()
+        response = client.get('/house/')
+        assert response.status_code == 200
```
Comment (Yarboa) [inline tests/my_house_tests.py:66]: Generic comment, I would consider checking the type of the template here

File: tests/my_house_tests.py [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+    def test_get_add_expense_view(self, client, new_user):
+        client.force_login(new_user)
+        HouseFactory(user=new_user).save()
```
Comment (Yarboa) [inline tests/my_house_tests.py:70]: Generic comment,
Consider using fixture the reyurn the HouseFactory
test_get_add_expense_view(self, client, new_user, house_facory):

**Review by Yarboa — CHANGES_REQUESTED** [review]
See, the comments,
You can apply it on all tests

File: tests/my_house_tests.py (hunk 7 above)
Comment (ShellyGolden) [reply tests/my_house_tests.py]: fixed

File: tests/my_house_tests.py (hunk 9 above)
Comment (ShellyGolden) [reply tests/my_house_tests.py]: fixed
